pkg/cable/wireguard/getconnections.go

```diff
@@ -2,10 +2,115 @@ package wireguard
 
 import (
 	"fmt"
+	"strconv"
+	"time"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+	"k8s.io/klog"
 
 	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
 )
 
 func (w *wireguard) GetConnections() (*[]v1.Connection, error) {
-	return nil, fmt.Errorf("Not implemented yet")
+	d, err := w.client.Device(DefaultDeviceName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find device %s: %v", DefaultDeviceName, err)
+	}
+	connections := make([]v1.Connection, 0)
+
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	for _, p := range d.Peers {
+		key := p.PublicKey
+		cid, err := w.clusterIDByPeer(&key)
+		if err != nil {
+			klog.Warningf("Found unknown peer with key %s, removing", key)
+			if err2 := w.removePeer(&key); err2 != nil {
```

Comment:
The purpose of this function is to retrieve current connection info thus it doesn't seem appropriate to mutate internal state as a side effect.

Perhaps the approach here should be the opposite, ie iterate the cached `Connection`s and retrieve its peer from `d.Peers` and , if not found, then report that fact in the returned `Connection`. I don't know how an orphaned connection could occur or if it would be a bug somewhere

Reply:
The source of truth should be the wg network device. 
No connection would be returned for submariner endpoints without a wg peer.
wg peers without a submariner endpoint indicate an error or someone changing things externally. Although I agree that this may not be appropriate for a status retrieval function, I think it is better to remove the wg peer.

Reply:
I think this applies to both ipsec and wg -- 
there are two "orphan" cases (1) a submariner endpoint without a corresponding cable connection (2) a cable connection without a submariner endpoint. For (1), do we return only existing cable connections or do we create a connection for each submariner endpoint, but with ERROR status -- I implemented the former. For (2), there is no other place where this is checked so either we report an error or delete the orphan cable connections -- I implemented the latter.